### PR TITLE
Replace PollOpErrorCode and IsUserError with CodeForError

### PR DESCRIPTION
**Pull request by amacaskill**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug

 /kind cleanup

> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Replace PollOpErrorCode and IsUserError with CodeForError, and filter out User errors from ControllerExpandVolume.This makes things more concise, and closely models what is done in the PDCSI driver. This will be followed up by a PR that filters out the rest of the context / user errors from the multishare controller. 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

### CODE REVIEW

File: pkg/cloud_provider/file/file.go [hunk 1]
```diff
@@ -681,15 +681,22 @@ func IsContextError(err error) *codes.Code {
...
+// (2) "context canceled", returns grpc Canceled
+func CodeForError(err error) *codes.Code {
+	if errCode := isUserError(err); errCode != nil {
 		return errCode
 	}
-	if errCode := IsContextError(err); errCode != nil {
+	if errCode := isContextError(err); errCode != nil {
 		return errCode
 	}
 	return util.ErrCodePtr(codes.Internal)
```
Comment (saikat-royc) [inline pkg/cloud_provider/file/file.go:702]: Can we also check if there is already some known status.code set in the err and skip overwrite of code to Internal.

if s, ok := status.FromError(test.err); ok {
 ... if s.Code() == // one of [codes](https://github.com/grpc/grpc-go/blob/v1.55.0/codes/codes.go)
}

Comment (amacaskill) [reply pkg/cloud_provider/file/file.go:720]: Done!

**Comment by judemars** [comment]
/cherry-pick release-1.4

(use this one not the one above, I got confused and deleted that one)

**Pull request merged**